src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -843,15 +847,29 @@ private InternalAggregations reduceAggregations(List<PercolateShardResponse> sha
             return null;
         }
 
+        InternalAggregations aggregations;
         if (shardResults.size() == 1) {
-            return shardResults.get(0).aggregations();
-        }
-
+            aggregations = shardResults.get(0).aggregations();
+        } else {
```

Comment:
Maybe we could simplify by only keeping the else part? I'm also a bit concerned that not calling reduce is wrong if the aggregation does more than merging results in reduce() (such as applying min_doc_count)